### ISSUE TITLE
0xC0000005 exception

### ISSUE THREAD

**Issue by lionpham** [MicrosoftDX/MicrosoftHLSSDK#15]
Hello,

I got 0xC0000005 exception when trying to open a HLS content. Bellowing is detailed information of this exception:

OS: Windows Phone 8.1
Exception Code: 0xC0000005
Exception Information:  The thread tried to read from or write to a virtual address for which it does not have the appropriate access.

Call Stack:
Microsoft.HLSClient.dll!Microsoft::HLSClient::Private::StreamInfo::DownloadPlaylistAsync(Concurrency::task_completion_event<long> tcePlaylistDownloaded) Line 215   C++
    Microsoft.HLSClient.dll!Microsoft::HLSClient::Private::Playlist::DownloadVariantStreamPlaylist(unsigned int desiredbitrate=104, bool failfast=104, short searchdirectionincaseoffailure=0, bool TestSegmentDownload=false) Line 292 C++
    Microsoft.HLSClient.dll!Microsoft::HLSClient::Private::Playlist::InitializePlaylistRefreshCallback::__l86::<lambda>() Line 862  C++
    Microsoft.HLSClient.dll!Platform::Details::__abi_FunctorCapture<<lambda_f4e2e657afc10a2e0c04311bb73fd675>,void,Windows::System::Threading::ThreadPoolTimer ^>::Invoke(Windows::System::Threading::ThreadPoolTimer ^ <__args_0>=0xfaf54068) Line 875 C++
    Microsoft.HLSClient.dll!Windows::System::Threading::TimerElapsedHandler::[Windows::System::Threading::TimerElapsedHandler::__abi_IDelegate]::__abi_Windows_System_Threading_TimerElapsedHandler___abi_IDelegate____abi_Invoke(Windows::System::Threading::ThreadPoolTimer ^ __param0=0xfaf54072)    C++
    threadpoolwinrt.dll!Windows::System::Threading::CThreadPoolTimer::TimerCallback(struct _TP_CALLBACK_INSTANCE *,void *,struct _TP_TIMER *)   Unknown
    ntdll.dll!TppTimerpExecuteCallback()    Unknown
    ntdll.dll!TppWorkerThread() Unknown
    ntdll.dll!RtlUserThreadStart()  Unknown

Looking forward to your help!

Best Regards

**Comment by ericleigh007** [comment]
I don't think that they'll be able to do anything without a code snippet, and probably the content that you tried to load.

**Comment by ngducnghia** [comment]
I have same issue when streaming HLS stream. Doesn't happens all the time but randomly during a stream:
Faulting module name: Microsoft.HLSClient.dll, version: 0.0.0.0, time stamp: 0x55e70c2e
Exception code: 0xc0000005
Fault offset: 0x00000000000fb5d4
Faulting process id: 0x1e4c
Faulting application start time: 0x01d1f10fb0123c29

Cause app to crash so regularly.

**Comment by jitghosh** [comment]
Hi – what version of Windows is this for ? If it is for Windows 10, then you should use the HLS capabilities built into the Windows API directly instead of this SDK. For more details – please look at https://msdn.microsoft.com/en-us/windows/uwp/audio-video-camera/adaptive-streaming .

**Comment by ngducnghia** [comment]
Hi Jitghosh,

Thanks for your quick reply. I'm developing Windows 8.1 application, but testing app on both Windows 8.1 and 10 has this issue. Is there any alternative for 8.1?
